web/yo/app/scripts/directives/curationQueue.js

```diff
@@ -212,46 +212,32 @@ angular.module('oncokbApp')
                         dueDay: $scope.input.dueDay ? new Date($scope.input.dueDay).getTime() : '',
                         comment: $scope.input.comment,
-                        notified: false
-                    });
+                        notified: false,
+                        hugoSymbol: hugoSymbol
+                    };
                     if ($scope.location === 'gene') {
-                        item.set('variant', $scope.input.variant);
+                        item['variant'] = $scope.input.variant;
                     } else if ($scope.location === 'queues') {
                         if ($scope.data.hugoVariantMapping[hugoSymbol]) {
-                            item.set('variant', $scope.data.hugoVariantMapping[hugoSymbol]);
+                            item['variant'] = $scope.data.hugoVariantMapping[hugoSymbol];
                         } else {
-                            item.set('variant', '');
+                            item['variant'] = '';
                         }
                     }
                     if ($scope.predictedArticle && $scope.validPMID) {
-                        item.set('article', $scope.predictedArticle);
-                        item.set('pmid', $scope.input.article);
+                        item['article'] = $scope.predictedArticle;
+                        item['pmid'] = $scope.input.article;
+                        item['pmidString'] = 'PMID: ' + item['pmid'];
                     } else {
-                        item.set('article', $scope.input.article);
-                    }
-                    $rootScope.queuesData.get(hugoSymbol).push(item);
-                    var queueItem = {
-                        article: item.get('article'),
-                        pmid: item.get('pmid'),
-                        pmidString: 'PMID: ' + item.get('pmid'),
-                        link: item.get('link'),
-                        variant: item.get('variant'),
-                        mainType: item.get('mainType'),
-                        subType: item.get('subType'),
-                        section: item.get('section'),
-                        addedBy: item.get('addedBy'),
-                        addedAt: item.get('addedAt'),
-                        curated: item.get('curated'),
-                        curator: item.get('curator'),
-                        dueDay: item.get('dueDay'),
-                        comment: item.get('comment'),
-                        notified: item.get('notified'),
-                        hugoSymbol: hugoSymbol
-                    };
-                    $scope.queue.push(queueItem);
-                    if (item.get('curator')) {
-                        $scope.sendEmail(queueItem);
+                        item['article'] = $scope.input.article;
                     }
+                    currentQueues.push(item);
+                    $scope.updateQueueInDB(hugoSymbol, currentQueues).then(function(result) {
```

Comment:
This is what I was talking about that everytime you change the queue content, you need to manually call the updateQueueInDB method. Isn't there a way to keep the current functionality?